### PR TITLE
Introduce ProcSubset, drop @clock @cpu-emulation

### PR DESCRIPTION
**Pull request by RZR7332**
Type of pull request:

- [X] Feature

This introduces the ProcSubset option and drops the @clock and @cpu-emulation system calls from the service.

ProcSubset was introduced in Systemd v247 and should present no issues following the bump to minimum v249.

@clock covers calls for manipulating the system clock and time (should not be required).
@cpu-emulation covers calls for CPU emulation functionality.

More details at https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html

### CODE REVIEW

File: data/fwupd.service.in [hunk 1]
```diff
@@ -18,6 +18,7 @@ MemoryDenyWriteExecute=yes
 NoNewPrivileges=no
 PrivateDevices=no
 PrivateTmp=true
+ProcSubset=pid
```
Comment (hughsie) [inline data/fwupd.service.in:21]: Does this allow us to read `/proc/1/cgroup`?

Comment (RZR7332) [reply data/fwupd.service.in:21]: I am not 100% certain. According to documentation:

subset=pid hides all top level files and directories in the procfs that are not related to tasks.

Source: https://docs.kernel.org/filesystems/proc.html#mount-options

Is there a way I can test this?

Comment (RZR7332) [reply data/fwupd.service.in:21]: After some further digging, it is also defined (default setting) as:

Service has full access to non-process /proc files (/proc subset=)

Comment (hughsie) [reply data/fwupd.service.in:21]: I think this means it's okay?

	sudo systemd-run  --wait -p ProcSubset=pid --pty --collect --service-type=exec cat /proc//1/cgroup
	Running as unit: run-u3717.service
	Press ^] three times within 1s to disconnect TTY.
	0::/init.scope
	Finished with result: success

Comment (RZR7332) [reply data/fwupd.service.in:21]: Looks readable to me with that output yes. ProcSubset is the setting I was most uncertain about in this PR.

**Review by hughsie — APPROVED** [review]

**Pull request merged**